### PR TITLE
Adds the Bluespace Drop Pod upgrade to the cargo express console

### PR DESCRIPTION
**Pull request by MrDoomBringer**
Researching an upgrade disk from RnD allows you to upgrade the Express Supply Console (#33436). 

Instead of normal drop pods, which crash then stick around until you dissassemble them, the console will call down new Bluespace Drop Pods, which will teleport out after crashing (and bring about a slightly smaller crash-explosion too!)

:cl: MrDoomBringer
add: The RnD department can now develop firmware upgrades to the Express Supply Console, unlocking advanced cargo drop pods!
/:cl:

**why: adds a little bit more techweb-integrated progression to cargo, also part of a  compromise from #34374.**

oh, also repaths bsdroppod.dm to supplypod.dm

### CODE REVIEW

**Comment by optimumtact** [comment]
this is in the cargo lathe right?

**Comment by MrDoomBringer** [comment]
yep

File: code/modules/cargo/expressconsole.dm [hunk 1]
```diff
@@ -24,6 +25,12 @@
 		locked = !locked
 		to_chat(user, "<span class='notice'>You [locked ? "lock" : "unlock"] the interface.</span>")
 
+	else if(istype(W, /obj/item/disk/cargo/bluespace_pod))
+		podID = 1//doesnt effect circuit board, so that reversal is possible
+		to_chat(user, "<span class='notice'>You insert the disk into the console, allowing for advanced supply delivery vehicles.</span>")
```
Comment (Cyberboss) [inline code/modules/cargo/expressconsole.dm]: into [src]

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -1354,7 +1354,7 @@ GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
 	desc = "A dusty SUPER-SIZED from the back of the Syndicate warehouse. Rumored to contain a valuable assortment of items, \
 			but you never know. Contents are sorted to always be worth 125 TC."
 	cost = 40
-	player_minimum = 40
+	//player_minimum = 40
```
Comment (Cyberboss) [inline code/modules/uplink/uplink_items.dm]: ditto

File: code/modules/uplink/uplink_items.dm [hunk 3]
```diff
@@ -1344,7 +1344,7 @@ GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
 			but you never know. Contents are sorted to always be worth 50 TC."
 	item = /obj/structure/closet/crate
 	cost = 20
-	player_minimum = 25
+	//pplayer_minimum = 25
```
Comment (Cyberboss) [inline code/modules/uplink/uplink_items.dm]: don't comment out code

File: code/game/objects/structures/crates_lockers/closets/secure/bsdroppod.dm [hunk 4]
```diff
@@ -1,87 +0,0 @@
-//The "BDPtarget" temp visual is created by the expressconsole, which in turn makes two things: a falling droppod animation, and the droppod itself.
```
Comment (Cyberboss) [inline code/game/objects/structures/crates_lockers/closets/secure/bsdroppod.dm]: Revert the line ending changes

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
You may as well have it roundstart if you're going to give it a glass-only material requirement and have it a part of the research generally done immediately at the first few minutes of the round.

Just Revert that PR instead of doing this please or actually make it be an upgrade /w investment. As it is, it's just silly.

**Comment by MrDoomBringer** [comment]
@ExcessiveUseOfCobblestone, I put it there because its the only node related to cargo. Techwebs balance is a bit out of my grasp, so if someone who has experience with balance wants to move it i'd be more than happy!

File: code/modules/uplink/uplink_items.dm (hunk 3 above)
Comment (MrDoomBringer) [reply code/modules/uplink/uplink_items.dm]: oh shit hahaha i was dicking around with supersurplus crates earlier this shouldnt have gotten through

File: code/game/objects/structures/crates_lockers/closets/secure/bsdroppod.dm (hunk 4 above)
Comment (MrDoomBringer) [reply code/game/objects/structures/crates_lockers/closets/secure/bsdroppod.dm]: line ending change? i repathed the name (even though it wasnt strictly necessary, i guess). @Cyberboss

**Comment by Iamgoofball** [comment]
@MrDoomBringer Dont ever worry about mineral reqs, cobby doessnt know what he's talking about. Time gates are the only thing that matters.

**Comment by ExcessiveUseOfCobblestone** [comment]
> roundstart rushes part tech to upgrade ORM and mining tech so miners will send minerals

This is like what most sci players do.

@Iamgoofball minerals are a timegate based heavily around RNG.

**Comment by Iamgoofball** [comment]
So it's a timegate

**Comment by Iamgoofball** [comment]
The research anyways

**Comment by Iamgoofball** [comment]
Also make the emagged ones force a no bluespace for that extra damage

**Comment by MrDoomBringer** [comment]
@Iamgoofball eh, emag should use whatever the console uses. Gives the player more options for effecting the station.

File: tgstation.dme [hunk 5]
```diff
@@ -1055,7 +1055,7 @@
 #include "code\game\objects\structures\crates_lockers\closets\utility_closets.dm"
 #include "code\game\objects\structures\crates_lockers\closets\wardrobe.dm"
 #include "code\game\objects\structures\crates_lockers\closets\secure\bar.dm"
-#include "code\game\objects\structures\crates_lockers\closets\secure\bsdroppod.dm"
+#include "code\game\objects\structures\crates_lockers\closets\secure\supplypod.dm"
```
Comment (ShizCalev) [inline tgstation.dme]: sort please. :(

Comment (Iamgoofball) [reply tgstation.dme]: its the dme dude

**Comment by Iamgoofball** [comment]
You're an idiot, Cobby. R&D isn't guaranteed to rush that tech. Stop forgetting anything can happen.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Iamgoofball people literally ask roundstart if miners want their tech if they don't rush it on their own accord.

Trying to look at the actual data it seems like it's **generally** researched EITHER ~10 or within ~30. That doesn't include the 3+ points you get to research at roundstart and given that advanced mining has prereqs iirc it is something short of rushed or given a short delay.

So yes, you're correct in that R&D isn't guaranteed to rush mining (something I never argued or said), but it does happen often enough to be a concern from what I saw in the actual techweb stats (trying to read that is awful btw).

Like I said earlier, if it's going to be inconsequential on all levels (research, mineral req, ease of upgrading the machine) most of the time, why bother making the initial change?

I want to be clear I do like the idea, I just wish it came off as more of an addition to the previous PR (an upgrade ingame) than a lame compromise of the previous PR (an arbitrary barrier that you kick down with no sweat and move on every other round).

**Comment by MrDoomBringer** [comment]
image

**Comment by MrDoomBringer** [comment]
in all seriousness, i think I agree with cobby here, especially making this a "milestone" for cargo (to some degree), rather than something that always gets researched every round. Not too sure on where a good node placement for it would be, though. (maybe make a new cargo node?)

**Pull request merged**